fluentd/generate_throttle_configs.rb

```diff
@@ -38,24 +47,75 @@ def get_file_name(name)
   return file_name
 end
 
+## Returns the names of all throttle configs for parsing through when reverting
+def get_all_throttle_files()
+  return Dir.glob('/var/log/es-container-*.log.pos')
+end
+
+def move_pos_file_project_entry(source_file, dest_file, project)
+
+  if File.file?(source_file)
+    project_pattern = ".*_#{project}_.*\.log"
+
+    matches = Array.new()
+    File.open(source_file) { |file|
+      file.grep(/#{project_pattern}/) { |match| matches << match }
+    }
+
+    update = ""
+    update = File.read(dest_file) if File.file?(dest_file)
+    matches.each { |match|
+      log_file_pattern = Regexp.escape(match.split(" ")[0])
+      update.gsub!(/#{log_file_pattern}.*$\n/, "")
+      ## We are clearing out the line while if it exists and then appending the source_file line
+      update << match
+    }
+    File.open(dest_file, "w") { |file| file.puts update }
+
+    ## Remove matches from the old file now that they're safely in the new file
+    update = File.read(source_file)
+    matches.each { |match|
+      log_file_pattern = Regexp.escape(match.split(" ")[0])
+      update.gsub!(/#{log_file_pattern}.*$\n/, "")
+    }
+    File.open(source_file, "w") { |file| file.puts update }
+  end
+
+end
+
+## This will copy the pos entries from the throttle configs back to the default pos file
+def revert_throttle()
+  get_all_throttle_files.each { |file_name| move_pos_file_project_entry(file_name, json_pos_file, '.*') }
+end
+
 def seed_file(file_name, project)
 
   if project.eql?('.operations')
     path = DEFAULT_OPS_PROJECTS.map{|p| get_project_pattern(p)}.join(',')
+
+    ## openshift-* is a protected project prefix -- so users would not be able to create
+    ## a project that starts with this. Guard taken to prevent possible collision with a
+    ## user created project 'operations'
+    pos_file = get_json_pos_file_name('openshift-operations')
   else
     path = get_project_pattern(project)
+    pos_file = get_json_pos_file_name(project)
```

Comment:
Is it possible that `project` could contain characters which are not valid in a filename?

Reply:
I don't believe so. It could be if a user enters it incorrectly in their config, otherwise the project names themselves are [:alnum:-_] which should be valid for file names